data-pipeline-ffi/src/trace_exporter.rs

```diff
@@ -111,8 +111,13 @@ pub unsafe extern "C" fn ddog_trace_exporter_send(
     trace_count: usize,
 ) -> MaybeError {
     // TODO - handle errors - https://datadoghq.atlassian.net/browse/APMSP-1095
+
+    // necessary that the trace be static for the life of the handle. Memory should be freed when
```

Comment:
Does this means that the caller has to make sure the `trace` memory is not freed or garbage collected for the life of the handle (quite similar to the life of the program). Should this be added to the doc of the function ?
Is the free handled by libdatadog ?

Reply:
This is a good callout and we should reword this comment (and update the doc comment). From the Trace Exporter's perspective we want `trace` to be static for the life of this function call, not necessarily for the entire life of the handle. If the caller wishes to re-use memory and swap new traces in to existing memory instead of re-allocating that is up to their implementation.